src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/UseExceptionThrowHelpers.cs

```diff
@@ -180,7 +180,8 @@ public override void Initialize(AnalysisContext context)
                     if (SymbolEqualityComparer.Default.Equals(objectCreationOperation.Type, ane))
                     {
                         if (aneThrowIfNull is not null &&
-                            IsParameterNullCheck(condition.Condition, out IParameterReferenceOperation? nullCheckParameter))
+                            IsParameterNullCheck(condition.Condition, out IParameterReferenceOperation? nullCheckParameter) &&
+                            nullCheckParameter.Type.TypeKind == TypeKind.Class)
```

Comment:
@Youssef1313, this won't be true for a generic constrained to `class` or some other reference type.  Is there a better way to check for that?

Reply:
maybe `nullCheckParameter.Type.IsReferenceType`?

Reply:
Perfect, thanks.